web/themes/custom/rebartik/templates/field--node--field-paragraph--article.html.twig

```diff
@@ -0,0 +1,95 @@
+ * to your theme's directory and customize it, just like customizing other
+ * Drupal templates such as page.html.twig or node.html.twig.
+ *
+ * Instead of overriding the theming for all fields, you can also just override
+ * theming for a subset of fields using
+ * @link themeable Theme hook suggestions. @endlink For example,
+ * here are some theme hook suggestions that can be used for a field_foo field
+ * on an article node type:
+ * - field--node--field-foo--article.html.twig
+ * - field--node--field-foo.html.twig
+ * - field--node--article.html.twig
+ * - field--field-foo.html.twig
+ * - field--text-with-summary.html.twig
+ * - field.html.twig
+ *
+ * Available variables:
+ * - attributes: HTML attributes for the containing element.
+ * - label_hidden: Whether to show the field label or not.
+ * - title_attributes: HTML attributes for the title.
+ * - label: The label for the field.
+ * - multiple: TRUE if a field can contain multiple items.
+ * - items: List of all the field items. Each item contains:
+ *   - attributes: List of HTML attributes for each item.
+ *   - content: The field item's content.
+ * - entity_type: The entity type to which the field belongs.
+ * - field_name: The name of the field.
+ * - field_type: The type of the field.
+ * - label_display: The display settings for the label.
+ *
+ *
+ * @see template_preprocess_field()
+ */
+#}
+{%
+  set classes = [
+  'field',
+  'field--name-' ~ field_name|clean_class,
+  'field--type-' ~ field_type|clean_class,
+  'field--label-' ~ label_display,
+  label_display == 'inline' ? 'clearfix',
+]
+%}
+{%
+  set title_classes = [
+  'field__label',
+  label_display == 'visually_hidden' ? 'visually-hidden',
+]
+%}
+
+{#       Image field settings       #}
+{%
+  set positions = {
+    'none': '',
+    'left': 'article-img-pos-left',
+    'right': 'article-img-pos-right'
+  }
+%}
+{%
+  set width = {
+    '30%': 'article-img-width-30p',
+    '40%': 'article-img-width-40p',
+    '50%': 'article-img-width-50p',
+    '100%': 'article-img-width-100p'
+  }
+%}
+
+{% if multiple %}
+  <div{{ attributes.addClass(classes, 'field__items') }}>
+    {% for item in items %}
+      {% set addon_classes = [] %}
+
+      {# Add position classes if specified in item `Position` field #}
+      {% if item.content['#paragraph'].field_position is defined and item.content['#paragraph'].field_position is not empty %}
+        {% set addon_classes = addon_classes|merge([positions[item.content['#paragraph'].field_position.value]]) %}
+      {% endif %}
+
+      {# Add width classes if specified in item `Width` field #}
+      {% if item.content['#paragraph'].field_width is defined and item.content['#paragraph'].field_width is not empty %}
+        {% set addon_classes = addon_classes|merge([width[item.content['#paragraph'].field_width.value]]) %}
+      {% endif %}
+
+      <div{{ item.attributes.addClass(classes, addon_classes, 'field__item') }}>{{ item.content }}</div>
```

Comment:
Craaaaaaazzzy. But interesting.

There is a much more simplest way, look:
```
{% set _class_position = 'article-img-pos-' ~ item.content['#paragraph'].field_position.value %}
{% set _class_width = 'article-img-width-' ~ item.content['#paragraph'].field_width.value %}

<div{{ item.attributes.addClass(classes, _class_position, _class_width, 'field__item') }}>{{ item.content }}</div>
```
And as well you have to change the 'field_width' values to
30p|30%
40p|40%
50p|50%
100p|100%

that's all you need :)

Reply:
thx ;)